### ISSUE TITLE
Documentation 

### ISSUE THREAD

**Issue by qzhu2017** [qzhu2017/PyXtal#45]
Currently, a few items in the documentation need to be added/corrected

1, Regarding the use of molecule (how to read the molecule from the SMILES string?)
2, The use of random_cluster generation (the basic functionality and the list of supported points groups)
3, The updates on the use of symmetry class

**Comment by scottfredericks** [comment]
@qzhu2017 
I have updated the documentation. Since I never managed to get the SMILES format working, I have omitted this option. As I stated, you are able to define the molecule using SMILES, but the coordinates will not be calculated; I cannot figure out how to do so using the OpenBabel Python bindings alone. Right now, we have 4 (documented) options for specifying molecules:
1) by string/collection
2) by pymatgen Molecule object
3) by filename
4) by string (based on pymatgen Molecule.from_str). This interprets the string in xyz or gaussian format. I am actually not sure if anyone will have use for this functionality.

I updated the section dealing with Wyckoff positions to focus on the new Group class. I think it reads a bit easier now. Also, I added a section on the random_cluster class, as well as some notes about the choice of axes for point groups.

I have not yet added the water ice example with altered lattice, because I still need to implement options for varying the lattice. For now I will just allow definition by lattice parameters, by matrix, or by volume. Adding more specific options would require a complete rewrite of the generate_lattice functions, and would require conditionally checking which values have or have not already been defined.
